sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -251,6 +258,15 @@ size_t gnrc_lorawan_build_uplink(gnrc_lorawan_t *mac, iolist_t *payload,
     lw_hdr->addr = mac->dev_addr;
     lw_hdr->fctrl = 0;
 
+    lorawan_hdr_set_adr(lw_hdr,mac->mlme.adr);
+
+    /* Set `ADRACKReq` bit and ADR_ACK_CNT */
+    if (mac->mlme.adr) {
+        lorawan_hdr_set_adr_ack_req(lw_hdr,
+                                    mac->mlme.adr_ack_cnt >= CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT ?
+                                    true : false);
```

Comment:
```suggestion
                                    mac->mlme.adr_ack_cnt >= CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT);
```

`>=` already resolves to boolean